tests/integration_tests.rs

```diff
@@ -1312,3 +1312,14 @@ fn acknowledgements() {
         )
         .stderr("");
 }
+
+// Regression test for https://github.com/sharkdp/bat/issues/1709
+#[test]
+fn fail_if_unsupported_supported() {
```

Comment:
Thank you for taking the time to make a contribution!

Before a detailed review, it will be very useful to have a broad set of regression tests in place. The test you added is only for the positive case. We would need tests also for the negative case, i.e. when the exit code is not 0.

I am also a bit skeptical about the name `--fail-if-unsupported` because it does not make much sense in and of itself. Since the use case specifically is usage as a less input preprocessor, I think we can simply call the flag `--as-input-processor-for-less`. I think it is perfectly fine to have long names for "experimental" feature. If it turns out that the flag can by used more broadly, we can also broaden the flag to e.g. just `--as-input-preprocessor`.

It is better to start narrow and iteratively broaden, than to start broad and iteratively narrow. Because the latter is hard to do in a backwards compatible way.